controllers/config_map.go

```diff
@@ -224,31 +310,48 @@ func (r *ConfigMapReconciler) reconcileDelete(ctx context.Context, cluster *capa
 	return ctrl.Result{}, nil
 }
 
-func (r *ConfigMapReconciler) AddFinalizer(ctx context.Context, awsCluster *capa.AWSCluster) error {
-	originalCluster := awsCluster.DeepCopy()
-	controllerutil.AddFinalizer(awsCluster, Finalizer)
-	return r.Client.Patch(ctx, awsCluster, client.MergeFrom(originalCluster))
+func (r *ConfigMapReconciler) AddFinalizer(ctx context.Context, capiCluster *capi.Cluster) error {
+	originalCluster := capiCluster.DeepCopy()
+	controllerutil.AddFinalizer(capiCluster, Finalizer)
+	return r.Client.Patch(ctx, capiCluster, client.MergeFrom(originalCluster))
 }
 
-func (r *ConfigMapReconciler) RemoveFinalizer(ctx context.Context, awsCluster *capa.AWSCluster) error {
-	originalCluster := awsCluster.DeepCopy()
-	controllerutil.RemoveFinalizer(awsCluster, Finalizer)
-	return r.Client.Patch(ctx, awsCluster, client.MergeFrom(originalCluster))
+func (r *ConfigMapReconciler) RemoveFinalizer(ctx context.Context, capiCluster *capi.Cluster) error {
+	originalCluster := capiCluster.DeepCopy()
+	controllerutil.RemoveFinalizer(capiCluster, Finalizer)
+	error := r.Client.Patch(ctx, capiCluster, client.MergeFrom(originalCluster))
+
+	//check if there is an AWSCluster with the same name and namespace as the capiCluster and remove the finalizer. This enables the migration of the finalizer from CAPA Cluster CR to CAPI Cluster CR
+	awsCluster := &capa.AWSCluster{}
+	err := r.Client.Get(ctx, types.NamespacedName{
+		Name:      capiCluster.Name,
+		Namespace: capiCluster.Namespace,
+	}, awsCluster)
+	if err == nil {
```

Comment:
We want to notice hard errors (apart from NotFound):

```suggestion
        if client.IgnoreNotFound(err) != nil {
            return err
        }
	if err == nil {
```